WalletWasabi.Gui/Program.cs

```diff
@@ -58,10 +68,17 @@ private static void Main(string[] args)
 					BuildAvaloniaApp().StartShellApp("Wasabi Wallet", AppMainAsync, args);
 				}
 			}
+			catch (OperationCanceledException ex)
+			{
+				Logger.LogDebug(ex);
+			}
 			catch (Exception ex)
 			{
 				Logger.LogCritical(ex);
-				Global.CrashReporter.SetException(ex);
+				if (!cliException)
```

Comment:
What is the motivation for this? AFAIK, you will show Crash Reporter in the following cases:

* `Global = CreateGlobal();` throws an exception. 
* `BuildAvaloniaApp().StartShellApp("Wasabi Wallet", AppMainAsync, args);` throws an exception.

but not when

`runGui = ShouldRunGui(args);` throws an exception.

Reply:
I don't want to show any window if the user uses command-line tools. I don't like my own solution here so I will replace it somehow. My other idea was to create a special exception type like `CommandLineException`. Add a catch here and wrap the exception as an inner exception into `CommandLineException`: https://github.com/molnard/WalletWasabi/blob/1151df950024dbe7a65891bee3e93df8cc7af7ed/WalletWasabi.Gui/Program.cs#L112

Reply:
I would try to avoid nested try-catches here. I think it would be a good thing to actually separate creation of `Global` to its own try-catch and when it fails, deal with it. When it is OK, then you can continue. It would greatly simplify thinking about what needs to be done to end correctly.

Reply:
That would probably force you to add a new method but that would be a good thing too.

Reply:
I removed the CLI exception and use runGui to determine if the crash report dialog is required or not.